### PR TITLE
[MPICH] specify file systems

### PR DESCRIPTION
**Pull request by simonbyrne**
MPI-IO isn't building correctly, see https://github.com/JuliaParallel/MPI.jl/pull/339#issuecomment-575971632

This might fix it.

### CODE REVIEW

**Comment by giordano** [comment]
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1071&view=logs&j=67c4cf6b-ce01-5cd0-58e8-2140b98a75e2&t=ece16658-6c8b-5dd8-4e1b-b65818fd6bff&l=13742
```
[23:44:07] sed: bad regex '\+': Repetition not preceded by valid expression
[23:44:07] checking for pvfs2-config... notfound
[23:44:07] checking configured file systems... 
```

File: M/MPICH/build_tarballs.jl [hunk 1]
```diff
@@ -11,7 +11,7 @@ script = raw"""
 # Enter the funzone
 cd ${WORKSPACE}/srcdir/mpich-*
 
-./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp
+./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp --with-file-system=testfs+ufs+nfs
```
Comment (giordano) [inline M/MPICH/build_tarballs.jl]: In `src/mpi/romio/configure` there is
```sh
#
# if FILE_SYSTEM is not set above, use ufs and nfs as default
#
if test -z "$FILE_SYSTEM" ; then
    FILE_SYSTEM="ufs nfs"
fi

# no matter what, always build testfs
FILE_SYSTEM="testfs $FILE_SYSTEM"
```
so my understanding is that this should be
```suggestion
./configure --prefix=$prefix --host=$target --enable-shared=yes --enable-static=no --disable-dependency-tracking --disable-fortran --docdir=/tmp --with-file-system="testfs ufs nfs"
```
but this list list is already the default.  What do you think?

Comment (giordano) [reply M/MPICH/build_tarballs.jl]: I don't know, I did some tests locally, I couldn't find a way to convince `configure` to accept the file systems passed from the command line :confused:

**Comment by giordano** [comment]
I think their regexp in `src/mpi/romio/configure`
```sh
# check for valid file system
if test -n "$FILE_SYSTEM" ; then
   # if multiple filesystems are passed in, they are '+'-delimited
   # we could set the IFS to tokenize FILE_SYSTEM, but the FILE_SYSTEM env var
   # is used in multiple places in the build system: get rid of the '+'s so we
   # can use the 'for x in $FILE_SYSTEM ...' idiom
   FILE_SYSTEM=`echo $FILE_SYSTEM|sed -e 's/\+/ /g'`
```
is simply wrong --  well, this needs to fix in `configure.ac`.  If I replace `\+` with `+` it works for me without explicitly passing `--with-file-system` in.  For such a simple substitution I'd use `tr` instead of `sed`...

**Comment by giordano** [comment]
Or anyway `sed` in BusyBox doesn't like this regexp.  Again, `tr` is just perfect for this use case, why complicating everything with regexps?

**Comment by simonbyrne** [comment]
Want to open an issue/PR on their repo: https://github.com/pmodels/mpich?

**Comment by giordano** [comment]
I've added the patch I submitted upstream: https://github.com/pmodels/mpich/pull/4289

**Comment by giordano** [comment]
It seems [it did the trick](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=1082&view=logs&j=8ad340e3-1b08-5604-897b-cc62f3641d2f&t=adb8966d-9bb5-54da-3c30-6af5dee699c7&l=13770):
```
[03:37:30] checking configured file systems... testfs ufs nfs
```

**Comment by simonbyrne** [comment]
Nice!

**Comment by ViralBShah** [comment]
Nice!

**Comment by giordano** [comment]
For the record, https://github.com/pmodels/mpich/pull/4289 has been merged, so from next release we won't need to apply the patch.

**Pull request merged**